Source/objects.cpp

```diff
@@ -1128,9 +1128,9 @@ void AddBarrel(Object &barrel)
 	barrel._oVar3 = GenerateRnd(3);
 
 	if (barrel._oVar2 >= 8) {
-		Monster *skeleton = PreSpawnSkeleton();
-		if (skeleton != nullptr) {
-			barrel._oVar4 = skeleton->getId();
+		auto maybeSkeleton = PreSpawnSkeleton();
+		if (maybeSkeleton) {
+			barrel._oVar4 = maybeSkeleton.value()->getId();
 		} else {
 			barrel._oVar4 = -1;
 		}

```

Comment:
What about `barrel._oVar4 = maybeSkeleton.value_or(-1);`?

Reply:
If I'm understanding you correctly here this would also require `SpawnSkeleton()` be changed to take an index intead of a pointer.